src/dom.ts

```diff
@@ -0,0 +1,134 @@
+// Positions used for node placement
+export enum Position {
+    LastIn,
+    FirstIn,
+    After,
+    Before,
+    Replace
+}
+
+// Tag trees for element creation
+let tagWrap: {[key: string]: any} = {
+	option: ['select'],
+	tbody: ['table'],
+	thead: ['table'],
+	tfoot: ['table'],
+	tr: ['table', 'tbody'],
+	td: ['table', 'tbody', 'tr'],
+	th: ['table', 'thead', 'tr'],
+	legend: ['fieldset'],
+	caption: ['table'],
+	colgroup: ['table'],
+	col: ['table', 'colgroup'],
+	li: ['ul']
+};
+
+for(let param in tagWrap){
+    let tw = tagWrap[param];
+    tw.pre = param == 'option' ? '<select multiple="multiple">' : '<' + tw.join('><') + '>';
+    tw.post = '</' + tw.reverse().join('></') + '>';
+}
+
+function insertAfter(node: Node, relativeElement: Node) {
+	let parent: Node = relativeElement.parentNode;
+	if (!parent) { return; }
+	if (parent.lastChild == relativeElement) {
+		parent.appendChild(node);
+	} else {
+		parent.insertBefore(node, relativeElement.nextSibling);
+	}
+}
+
+/**
+ * Places a node in the DOM relative to another node
+ *
+ * @param node The node to place in the DOM
+ * @param position The position to place the node, relative to relativeElement
+ * @param relativeElement The node to use as a reference when placing
+ *
+ * @example
+ * dom.place(node, anotherNode, dom.Position.After);
+ */
+export function place<T extends Node>(node: T, position: Position, relativeElement: Element): void {
+	let parent: Node = relativeElement.parentNode;
+
+	if (!parent) {
+		throw new Error('Reference node must be in DOM');
+	}
+
+	switch (position) {
```

Comment:
Consider changing this to if/else; there isn't much reason for a switch here (other than not repeating `position` each time)